### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by jorshi**
Starting unit tests with PyTest

### CODE REVIEW

File: src/ddspdrum/module.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __repr__(self):
         )
 
     def seconds_to_samples(self, seconds: float) -> int:
-        return round(int(seconds * self.sample_rate))
+        return int(round(seconds * self.sample_rate))
```
Comment (jorshi) [inline src/ddspdrum/module.py:48]: I think we want to round before converting to int here

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -42,3 +44,18 @@ jobs:
         fail_ci_if_error: true
 #        path_to_write_report: ./coverage/codecov_report.txt
         verbose: true
+    - name: Generate coverage report with tests
```
Comment (jorshi) [inline .github/workflows/codecov.yml:47]: Adding pytest to codecov report

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Tests with Pytest
```
Comment (jorshi) [inline .github/workflows/pytest.yml:1]: maybe redundant adding a pytest specific workflow? But we can add tests for different OSes and python versions in addition to the one run in the code coverage

File: tests/test_module.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+"""
```
Comment (jorshi) [inline tests/test_module.py:1]: pytest looks for modules prefixed with test_ and then runs methods prefixed with test_ as unit tests. Grouping each test within a class that relates to the class that it is testing.

**Review by turian — APPROVED** [review]

**Pull request merged**